src/new-components/modal/modal.tsx

```diff
@@ -248,13 +248,15 @@ export function ModalHeader({
             >
                 <Columns space="large" alignY="center">
                     <Column width="auto">{children}</Column>
-                    <Column width="content" exceptionallySetClassName={styles.buttonContainer}>
-                        {typeof button !== 'boolean' ? (
-                            button
-                        ) : button === true ? (
-                            <ModalCloseButton aria-label="Close modal" autoFocus={false} />
-                        ) : null}
-                    </Column>
+                    {button === false || button === null || button === undefined ? null : (
```

Comment:
Can this be a single falsy check instead?

Reply:
I opted not to use it. The reason is that `0` should be a valid button content.

I could have written `button === false || button == null`, but then it triggers an eslint rule, so I wrote the full statement.

Reply:
Indeed. Technically it is. That being said, technically I do not mind if we'd overlook that. Another reason for a falsy check would be to also consider the empty string as there being no button to show, which IMO is valid. But I have no strong opinion about having to do something about it right now.